src/tree.rs

```diff
@@ -1940,3 +1986,213 @@ pub enum Content {
     Folder { title: String },
     Separator { position: i64 },
 }
+
+/// Completion operations to apply to the local tree after a merge. These are
+/// represented as separate structs in `Vec`s instead of enums yielded from an
+/// iterator so that consumers can easily chunk them. Places writes these ops
+/// into temp tables, then runs SQL triggers to apply them.
+#[derive(Clone, Debug, Default)]
+pub struct CompletionOps<'t> {
```

Comment:
I'm not sure if there's a better way to do this...the idea is to have a temp table per op (`CREATE TEMP TABLE changeGuidOps(localGuid TEXT PRIMARY KEY, mergedGuid TEXT NOT NULL)`, `applyRemoteOps(mergedGuid, remoteGuid)`, and so on).

I also tried having a `CompletionOp` with a set of flags per item, and keeping them all in a single `completionOps` table. This would be similar to how the `mergeStates` table works today, except it would only hold changed items, instead of all items in the tree. But that'd mean we need to do 3-4 table scans on `completionOps`, since the ops need to happen sequentially: we need to change _all_ GUIDs before we apply remote items, and insert _all_ items before applying structure and updating change counters. We can't do this per item because an item might, for example, reference a parent GUID that we haven't inserted yet...and, even though we can guarantee level order in the list of ops, SQLite won't necessarily fire triggers in the same order.

Reply:
It's a little verbose, but I think that's OK - explicit is good!